content/document.ts

```diff
@@ -34,7 +34,7 @@ import {
   MEMOIZE_INVALIDATE,
 } from "./utils.js";
 import * as Redirect from "./redirect.js";
-import { DocFrontmatter } from "../libs/types/document.js";
+import { DocFrontmatter, UnbuiltDocument } from "../libs/types/document.js";
```

Comment:
Maybe also worth moving `BuiltDocument` into `libs/types` if it isn't already, and extracting common parts if any?

Reply:
Never mind for now.